acorn/src/statement.js

```diff
@@ -148,6 +191,18 @@ pp.parseStatement = function(context, topLevel, exports) {
       return this.parseFunctionStatement(node, true, !context)
     }
 
+    let usingKind = this.isAwaitUsing(false) ? "await using" : this.isUsing(false) ? "using" : null
+    if (usingKind) {
+      if (topLevel && this.options.sourceType === "script") {
+        this.raise(this.start, "Using declaration cannot appear in the top level when source type is `script`")
+      }
+      this.next()
+      if (usingKind === "await using") this.next()
```

Comment:
I would think that `{await using a = x}` in `sourceType: script` would be a syntax error, but it doesn't seem to be the case currently. I think maybe we should check `this.canAwait` here.

Reply:
Thank you! fixed: https://github.com/acornjs/acorn/pull/1369/commits/e59ede74b098dc3276bd68f918de441f237f68f1